### PR TITLE
Use pipenv

### PR DESCRIPTION
**Pull request by dethos**
This pull request is an experiment. It intends to work as an example on how switching to `pipenv` would look like, to help in the discussion of #278. 

Also it handles part of the #81 issues as well, since the readme was updated to the new workflow using `pipenv`.

**Note: CI will fail, since it isn't yet prepared to this new workflow.

### CODE REVIEW

**Comment by dethos** [comment]
CI was updated to use `pipenv`. This pull request should be ready for review/merge.

**Comment by dethos** [comment]
@tomascarvalho any news about this?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -8,23 +8,15 @@ Below are a few things we follow and would appreciate if you do too.
...
+or in case it is only useful for developers add the `--dev` flag.
 
-and 
-
-```
-$ pip-compile requirements/base.in requirements/development.in -o requirements/requirements_dev.txt
-```
-
-* Commit these changes alongside your other code modifications
+You should add the changes in `Pipfile` and `Pipfile.lock` to your commit.
```
Comment (tomascarvalho) [inline CONTRIBUTING.md:19]: Should explain how to lock changes to `Pipfile.lock` with `pipenv lock` and `pipenv lock --dev`?

File: README.md [hunk 2]
```diff
@@ -27,41 +27,28 @@ Base requirements:
 * Python 3
 * Redis
 * PostgreSQL
-* gulp
 
-## Linux
+## On Linux
```
Comment (tomascarvalho) [inline README.md:31]: Should we explicitly mention Mac OS?

**Review by tomascarvalho — COMMENTED** [review]
Seems ok.
Made a few comments.

File: CONTRIBUTING.md (hunk 1 above)
Comment (dethos) [reply CONTRIBUTING.md:19]: @tomascarvalho `pipenv install` already adds the new information to the `.lock` file.

File: README.md (hunk 2 above)
Comment (dethos) [reply README.md:31]: It should be very similar now with the new `pipenv` flow. Nevertheless, I will only support linux for now (I don't have a mac hardware to support it). We might add it again if there is sufficient interest by future contributors to the project.

Comment (tomascarvalho) [reply README.md:31]: I have mac and it works just fine with the Linux instructions. But I guess we can see about this later.

**Review by tomascarvalho — APPROVED** [review]

**Pull request merged**